packages/starknet-snap/src/types/snapApi.ts

```diff
@@ -55,6 +56,9 @@ export type ApiRequestParams =
 export type BaseRequestParams = {
   chainId?: string;
   debugLevel?: string;
+  transactionVersion?:
+    | typeof constants.TRANSACTION_VERSION.V2
+    | typeof constants.TRANSACTION_VERSION.V3;
```

Comment:
i think it should be applied in some of the request only, but not all (even it is optional)